src/data/fun_like/fintype.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2022 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import data.finite.basic
+import data.fintype.basic
+import data.fun_like.basic
+
+/-!
+# Finiteness of `fun_like` types
+
+We show a type `F` with a `fun_like F α β` is finite if both `α` and `β` are finite.
+This corresponds to the following two pairs of declarations:
+
+ * `fun_like.fintype` is a definition stating all `fun_like`s are finite if their domain and
+   codomain are.
+ * `fun_like.finite` is a lemma stating all `fun_like`s are finite if their domain and
+   codomain are.
+ * `fun_like.fintype'` is a non-dependent version of `fun_like.fintype` and
+ * `fun_like.finite` is a non-dependent version of `fun_like.finite`, because dependent instances
+   are harder to infer.
+
+You can use these to produce instances for specific `fun_like` types.
+(Although there might be options for `fintype` instances with better definitional behaviour.)
+They can't be instances themselves since they can cause loops.
+-/
+
+variables (F G : Type*) {α γ : Type*} {β : α → Type*} [fun_like F α β] [fun_like G α (λ _, γ)]
```

Comment:
```suggestion
variables (F G α γ : Sort*} {β : α → Sort*} [fun_like F α β] [fun_like G α (λ _, γ)]
```

Reply:
This one doesn't work because `fintype` is defined on `Type`, which in retrospect should have been obvious to me :)